credentials: sample delegated crl signer certs and CRLs

shubhamdp opened the pull request:
Generated the samples as described in https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10308

Added README file which describes the content of the directory.

Its a follow up on #36838, and ticks all items from https://github.com/project-chip/connectedhomeip/issues/26582#issuecomment-2558952413.

Fixes #26582

Generated the delegated CRL signers using the cryptography module from python.

### Testing
- This PR is the building block for automating the DA revocation implementation. It adds the test data for generating the device attestation revocation set.
- Newly added test data is manually tested by dumping the certs/kets/crl using openssl [x509 | ec | crl] command. Also verified if both .der and .pem pairs outputs the same data.

shubhamdp committed:
credentials: sample delegated crl signer certs and CRLs

Generated the samples as described in
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10308

restyled-commits committed:
Restyled by prettier-markdown

andy31415 reviewed (approved):
@shubhamdp why are adding any form of

shubhamdp commented:
@tcarmelveilleux, @bh3000 can you guys please review this PR.

shubhamdp commented:
> @shubhamdp why are adding any form of

@andy31415 could you please elaborate on this.

shubhamdp committed:
fix some paths in the readme file

tcarmelveilleux commented on credentials/test/revoked-attestation-certificates/README.md:
```diff
@@ -0,0 +1,67 @@
+used for testing purposes only and should not be used in production
+environments.
+
+## Direct CRL Signing
```
@bh3000 Can you please review this?

bh3000 replied:
Done

bh3000 commented on credentials/test/revoked-attestation-certificates/indirect/Chip-Test-PAA-FFF1-Delegated-CRL.pem:
This has an Issuing Distribution Point with a URI pointed to a .pem file. The spec calls out that the dataUrl must point to  a DER file and must match this URI byte-for-byte (See 11.23.9.8. DataUrl).  Can you update to DER encoded binary file?

bh3000 replied:
Same is true for the PAI CRL.

bh3000 commented on credentials/test/revoked-attestation-certificates/indirect/Chip-Test-PAA-FFF1-Delegated-CRL.pem:
Since this is a delegated signer, it is an 'indirect CRL'.  Spec requires that the indirectCRL field be set to true for these cases, but it is missing here. (See 11.23.9.8. DataUrl)

shubhamdp replied:
Set the `indirect CRL` field to True.

bh3000 commented:
I think you're only missing one scenario from your checklist: CRL with Certificate Issuer CRL entry extension, only first entry has the extension.

shubhamdp committed:
address reviews

uri should point to a .der file and .pem
Set the indirectCRL field to true
Also adding the crl with Certificate Issuer CRL entry extension, only first entry has the extension.

restyled-commits committed:
Restyled by prettier-markdown

shubhamdp commented:
> I think you're only missing one scenario from your checklist: CRL with Certificate Issuer CRL entry extension, only first entry has the extension.

Added another CRL for this purpose: Chip-Test-PAI-FFF1-Delegated-CRL-Only-First-Entry-With-Issuer-Ext.pem

shubhamdp committed:
Added few more test data set

- Added DAC certs signed by revoked PAI, out of it one DAC is revoked
  and one is not.
- Added the dac provider test vectors for the indirect crl's usecase
- Added the revocation set for the indirect crl's

restyled-commits committed:
Restyled by prettier-json

shubhamdp commented:
@bh3000 can you please take a look.
